support/sier-codec/src/json.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{Error, Object, StructDef, Type, Value};
+use serde_json::Value as SerdeValue;
+use std::sync::Arc;
+
+pub fn transform_serde_value<'a>(
+    value: &SerdeValue,
+    type_: &'a Type,
+) -> Result<Value<'a>, Error<'a>> {
+    match value {
+        SerdeValue::Null => unimplemented!(),
+        SerdeValue::Bool(_bool) => unimplemented!(),
+        SerdeValue::Number(number) => {
+            if type_ == &Type::U64 {
+                if let Some(n) = number.as_u64() {
+                    Ok(Value::U64(n))
+                } else {
+                    unimplemented!()
+                }
```

Comment:
This can be `number.as_u64().ok_or_else(|| unimplemented!())`.

Reply:
But we need to wrap the `u64` into `Value::U64` and the ok_or_else returns a `Result`. Can you please provide an example, because I don't it would look cleaner this way.